src/Microsoft.AspNet.Mvc.Core/ModelBinding/Validation/DefaultObjectValidator.cs

```diff
@@ -166,32 +152,49 @@ private void MarkPropertiesAsSkipped(string currentModelKey, ModelMetadata metad
             }
         }
 
-        private bool ValidateProperties(
+        private IList<IModelValidator> GetValidators(IModelValidatorProvider provider, ModelMetadata metadata)
+        {
+            var validatorProviderContext = new ModelValidatorProviderContext(metadata);
+            provider.GetValidators(validatorProviderContext);
+            return validatorProviderContext.Validators;
+        }
+
+        private bool ValidateChildNodes(
             string currentModelKey,
             ModelExplorer modelExplorer,
             ValidationContext validationContext)
         {
             var isValid = true;
+            ExpandValidationNodeAndGetValidators(validationContext, modelExplorer);
 
-            foreach (var property in modelExplorer.Metadata.Properties)
+            IList<IModelValidator> validators = null;
+            if (modelExplorer.Metadata.IsCollectionType)
             {
-                var propertyExplorer = modelExplorer.GetExplorerForProperty(property.PropertyName);
-                var propertyMetadata = propertyExplorer.Metadata;
+                var enumerableModel = (IEnumerable)modelExplorer.Model;
+                var elementType = GetElementType(enumerableModel.GetType());
+                var elementMetadata = _modelMetadataProvider.GetMetadataForType(elementType);
```

Comment:
>in what scenario is metadata missing or incorrect?

It is not missing or incorrect, the propertyMetadata represents IEnumerable<T> we want the metadata for T.

>seems to duplicate ModelMetadata determination that should be done when each child node was originally created (in model binder or ExpandValidationNodeAndGetValidators)

Its there in child node, but that would mean to have this check repeated for every child node. and the code to do that would be slightly complex.
Not fixing it. If we have perf issues .. will fix later.